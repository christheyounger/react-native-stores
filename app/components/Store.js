import React from 'react';
import { FlatList, ActivityIndicator, Text, View  } from 'react-native';

export default class Store extends React.Component {
  render() {
    return (
      <View>
        <Text>- { this.props.name }</Text>
        <FlatList style={{paddingLeft: 10}}
          data={this.props.branches}
          renderItem={({item}) => <Store name={item.name} branches={item.branches} />}
          keyExtractor={({id}, index) => id.toString()}
        />
      </View>
    )
  }
}
