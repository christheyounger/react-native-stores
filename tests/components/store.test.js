import React from 'react';
import Store from '../../app/components/Store';
import renderer from 'react-test-renderer';

describe('component.Store', function() {
  const store = {id: 12, name: 'Root Store', branches: [{id: 14, name: 'Child Store', branches: []}]};
  const wrapper = renderer.create(
    <Store
      id={store.id}
      name={store.name}
      branches={store.branches}
      handleClick={jest.fn}
    />
  ).toJSON();
  test('renders correctly', function() {
    expect(wrapper).toMatchSnapshot();
  });
});
