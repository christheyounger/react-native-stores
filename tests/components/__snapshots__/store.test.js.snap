// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component.Store renders correctly 1`] = `
<View>
  <Text
    onPress={[Function]}
  >
    - 
    Root Store
  </Text>
</View>
`;
